core/src/test/java/io/questdb/test/cairo/mv/MatViewTest.java

```diff
@@ -3395,6 +3579,123 @@ private void testIncrementalRefresh0(String viewSql) throws Exception {
+                            "price_1h\tincremental_interval\tbase_price\t" + initialClock + "\t" + initialClock + "\tselect sym, last(price) as price, ts from base_price sample by 1h\tprice_1h~2\t\tvalid\t1\t1\t0\tHOUR\t" + start + "\t1\tHOUR\n",
+                    "materialized_views",
+                    null
+            );
+
+            execute("insert into base_price(sym, price, ts) values('jpyusd', 104.57, '2024-09-10T13:02')");
+
+            currentMicros += clockJump;
+            drainMatViewIntervalQueue(refreshIntervalJob);
+            drainQueues();
+
+            assertQueryNoLeakCheck(
+                    "sym\tprice\tts\n" +
+                            "gbpusd\t1.323\t2024-09-10T12:00:00.000000Z\n" +
+                            "gbpusd\t1.321\t2024-09-10T13:00:00.000000Z\n" +
+                            "jpyusd\t103.21\t2024-09-10T12:00:00.000000Z\n" +
+                            "jpyusd\t104.57\t2024-09-10T13:00:00.000000Z\n",
+                    "price_1h order by sym"
+            );
+        });
+    }
+
+    private void testIntervalRefreshMatViewSmallJumps(String start, String every, String initialClock, long clockJump, int ticksBeforeRefresh) throws Exception {
+        assertMemoryLeak(() -> {
+            execute(
+                    "create table base_price (" +
+                            "sym varchar, price double, ts timestamp" +
+                            ") timestamp(ts) partition by DAY WAL"
+            );
+            execute(
+                    "create materialized view price_1h refresh start '" + start + "' every " + every + " as (" +
+                            "select sym, last(price) as price, ts from base_price sample by 1h" +
+                            ") partition by day"
+            );
+
+            execute(
+                    "insert into base_price(sym, price, ts) values('gbpusd', 1.320, '2024-09-10T12:01')" +
+                            ",('gbpusd', 1.323, '2024-09-10T12:02')" +
+                            ",('jpyusd', 103.21, '2024-09-10T12:02')" +
+                            ",('gbpusd', 1.321, '2024-09-10T13:02')"
+            );
+
+            currentMicros = parseFloorPartialTimestamp(initialClock);
+            final MatViewRefreshIntervalJob refreshIntervalJob = new MatViewRefreshIntervalJob(engine);
+
+            for (int i = 0; i < ticksBeforeRefresh; i++) {
+                drainMatViewIntervalQueue(refreshIntervalJob);
+                drainQueues();
+
+                assertQueryNoLeakCheck(
+                        "sym\tprice\tts\n",
+                        "price_1h order by sym"
+                );
+                assertQueryNoLeakCheck(
```

Comment:
Can we also assert that refresh interval job does not trigger notifications to mv refresh jobs when timer is not expired?

Reply:
That's done in `MatViewTest#testIncrementalRefreshStatementOnTimerMatView()` and `MatViewReloadOnRestartTest#testTimerMatViewsReloadOnServerStart2()`